### ISSUE TITLE
Links to .md files are broken (runtime)

### ISSUE THREAD

**Issue by kptdobe** [adobe/helix-pipeline#472]
- Open [http://helix-pipeline-adobe.hlx.page/README.html](http://helix-pipeline-adobe.hlx.page/README.html)
- Check links (context, schema...)

-> all links to `.md` files are broken, the href ends with a `.`: `<a href="docs/context.schema.">context</a>` instead of `.html`.

This is true for other repos like [https://helix-example-basic-adobe.project-helix.page/](https://helix-example-basic-adobe.project-helix.page/).

**Comment by kptdobe** [comment]
So tricky...

This commit https://github.com/adobe/helix-pipeline/commit/2df18b97a83e38bacbd994fe06ee7fdc58539b6d introduced the regression. 

```
const extension = request && request.extension ? request.extension : 'html';
```

is not equivalent to 

```
const extension = setdefault(request, 'extension', 'html');
```

if extension is an empty string... (`request && request.extension` is false in that case)

In the second case, `setdefault` returns `extension` property from `request` if it is already present! Which is the case in the "runtime" case: see https://github.com/adobe/helix-pipeline/blob/efdc964320d08aaf1fd843659012476e3653de60/src/utils/openwhisk.js#L66 (empty string is set if not incoming `extension` parameter which never set by the dispatch action).

Now I need to find how to do the same with Ferrum :)

**Comment by tripodsan** [comment]
> (empty string is set if not incoming extension parameter which never set by the dispatch action).

so maybe this is wrong, then? since it used to be different.

**Comment by rofe** [comment]
Fixed in #473
